File: c_algorithms/sorting/bubble_r.c
#include "PPSH.h"

int main()
{
   int nums[] = {25, 15, 19, 48, 27, 42, 40, 13, 4, 30, 39, 33, 44, 16, 49, 14, 28, 17, 47, 31};

  int s1 = sizeof(nums)/sizeof(nums[0]);

  printarray(nums, s1);
  printf("\n");
  bool needed = 0;
  int passes = 0;
//The Sort
  do {
    needed = 0;
    for(int i=0; i<s1; i++){
      if (nums[i] > nums[i+1])
        {
          needed=1;
          swapints(&nums[i],&nums[i+1]);
        }
      }
      passes++;
    } while(needed);

  printf("Sorted Array:\t");
  printarray(nums, s1);
  printf("\n");
  printf("Number of passes: %d\n", passes);

  return 0;
}

File: python_algorithms/sorting/bubble_r.py
nums = [25, 15, 19, 48, 27, 42, 40, 13, 4, 30, 39, 33, 44, 16, 49, 14, 28, 17, 47, 31]

s1 = len(nums)
print("Number of elements in the array: %d" % s1)
print("Original Array: %a" % nums)

needed = True
passes = 0


while(needed):
    needed = False
    for i in range(0, s1-1):
        if nums[i] > nums[i+1]:
            needed = True
            tmp = nums[i]
            nums[i] = nums[i+1]
            nums[i+1] = tmp
    passes = passes + 1

print("Sorted Array: %a" % nums)
print("Number of passes: %d" % passes)
